src/averageAges.js

```diff
@@ -55,7 +63,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const kids = people.filter(({ mother, sex }) => !onlyWithSon
+    ? people.find(({ name }) => name === mother)
+    : sex === 'm' && people.find(({ name }) => name === mother));
+
+  const diff = kids.map(({ born, mother }) => born - momBorn(people, mother));
+
+  return diff.reduce((sum, age) => sum + age) / diff.length;
+}
+
+function momBorn(people, mother) {
```

Comment:
```suggestion
function getMomBorn(people, mother) {
```
we usually name functions as verbs